Report-870

niyolynate opened the pull request:
Still haven't sent my changes, i get errors..
i get errors in git while pushing them. will keep trying though.   i added restart @line45 - 49 and added time for tasks at line 90-99.


package org.openmrs.module.reporting;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.openmrs.api.context.Context;
import org.openmrs.module.BaseModuleActivator;
import org.openmrs.module.DaemonToken;
import org.openmrs.module.DaemonTokenAware;
import org.openmrs.module.reporting.common.MessageUtil;
import org.openmrs.module.reporting.report.task.ReportingTimerTask;
import org.openmrs.module.reporting.report.task.RunQueuedReportsTask;

/**
 * This class contains the logic that is run every time this module is either started or shutdown
 */
public class ReportingModuleActivator extends BaseModuleActivator implements DaemonTokenAware {

	private Log log = LogFactory.getLog(this.getClass());

    @Override
    public void contextRefreshed() {
		// This will ensure that the MessageUtil picks back up the current active message source
		MessageUtil.setMessageSource(null);
    }

    @Override
	public void started() {
		ReportingTimerTask.setEnabled(true);
		log.info("Reporting Module Started...");
	}

    @Override
    public void restart()
    restartAllScheduledTasks();
    restartCurrentRunningReportRequest;
}

	@Override
	public void willStop() {
		cancelAllScheduledTasks();
		cancelCurrentlyRunningReportRequests();
	}

	@Override
	public void stopped() {
		log.info("Reporting Module Stopped...");
	}

	@Override
	public void setDaemonToken(DaemonToken token) {
		ReportingTimerTask.setDaemonToken(token);
	}

	/**
	 * Cancels all scheduled tasks
	 */
	private void cancelAllScheduledTasks() {
		for (ReportingTimerTask task : Context.getRegisteredComponents(ReportingTimerTask.class)) {
			try {
				task.cancel();
			}
			catch (Exception e) {
				log.warn("An exception occurred while trying to stop reporting task " + task.getTaskClass().getSimpleName(), e);
			}
		}
	}
	/**
	 * Cancels all currently running report requests
	 */
	private void cancelCurrentlyRunningReportRequests() {
		for (RunQueuedReportsTask task : RunQueuedReportsTask.getCurrentlyRunningRequests().values()) {
			try {
				task.cancelTask();
			}
			catch (Exception e) {
				log.warn("An exception occurred while trying to stop currently running reports", e);
				 
				//Time lap between the last task to current updates.
				public voidsetTask(Integer time) {
					    oldT = t;
					    t = time;
					    log.debug("Time set to {}. innitialDate {}.", t, innitialDate);
					    if(time.intValue() > 50sec) {
					      log.info("Time passed in days\hr\min\sec.");
			}
		}
	}
}

Ben-Waters committed:
REPORT-714 : added a handler for CohortIndicators that filters out any other generic indicators (#164)

Ben-Waters committed:
REPORT-842 : Switch License to MPL v2.0 (#166)

mseaton committed:
REPORT-843
BirthAndDeathCohortDefinition should query on whether patient is marked as dead

openmrs-bot committed:
[maven-release-plugin] prepare release 1.16.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

samuelmale committed:
REPORT-420:Added attribute of viewing affected Report Defs on the UI on editing a particular dataSet

mseaton committed:
REPORT-845 - Support data set definition based on a SQL file and configurable connection

mseaton committed:
REPORT-845 - Support data set definition based on a SQL file and configurable connection

mogoodrich committed:
fix for surefire bug

openmrs-bot committed:
[maven-release-plugin] prepare release 1.17.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

Ruhanga committed:
REPORT-838 : Update Drug Order Filter

mseaton committed:
REPORT-849
PatientIdentifierDataEvaluator should return all identifiers if no identifier type specified

cioan committed:
REPORT-852, implement ObsOnSameDateEncounterDataDefinition

mseaton committed:
REPORT-852 - Implement Obs on same day EncounterDataDefinition
Additional enhancements to include ability to filter on coded answers

mogoodrich committed:
REPORT-853: Add no-args constructors to all converters

mogoodrich committed:
REPORT-854: Make SortCriteria SortElement inner class static

mogoodrich committed:
REPORT-854: Make inner classes static

mseaton committed:
REPORT-854: Add custom converter for SortCriteria in order to property deserialize before and after change to convert non-static inner class to static inner-class

mseaton committed:
Attempt to fix travis builds by moving to OpenJDK

mogoodrich committed:
Revert: REPORT-854: Add custom converter for SortCriteria in order to property deserialize before and after change to convert non-static inner class to static inner-class

mogoodrich committed:
Revert: REPORT-854: Make inner classes static

mogoodrich committed:
REPORT-854: Make inner classes serializable

openmrs-bot committed:
[maven-release-plugin] prepare release 1.18.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

kant committed:
Minor formatting (proposals) (#181)

mozzy11 committed:
REPORT-855 : Improved the BuiltInCohortDefinitionLibrary For Coded_obs and text_obs

mozzy11 committed:
REPORT-856: Create ConditonCohortDefinition class and add Tests

mozzy11 committed:
REPORT-856: ActiveOnDate should Include onset and End dates

mozzy11 committed:
Merge pull request #186 from mozzy11/REPORT-856-II

REPORT-856: ActiveOnDate should Include onset and End dates

mseaton committed:
REPORT-858: Follow-up commit to fix test errors when adding the 2.2 profile

mseaton committed:
REPORT-856 - follow-up fix required due to new Serialization Xstream dependency that has broken unit tests around ImplementerConfigured definitions.

mseaton committed:
REPORT-856 - Follow-up commit to resolve remaining 2.2 compatibility issues.

mseaton committed:
REPORT-856 - Follow-up commit to resolve remaining 2.2 compatibility issues.

mseaton committed:
REPORT-856 - Follow-up commit to resolve remaining 2.2 compatibility issues.

mseaton committed:
REPORT-856 - Follow-up commit to resolve remaining 2.2 compatibility issues.

mseaton committed:
REPORT-856 - Follow-up commit to resolve remaining 2.2 compatibility issues.

mseaton committed:
Update to the latest released version of testutils in preparation for release.

openmrs-bot committed:
[maven-release-plugin] prepare release 1.19.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

kant committed:
Minor formatting proposals (#190)

solemabrothers committed:
REPORT-859: Add a JSON Renderer to TextTemplate feature (#192)

JLLeitschuh committed:
Use HTTPS instead of HTTP to resolve dependencies (#194)

This fixes a security vulnerability in this project where the `pom.xml`
files were configuring Maven to resolve dependencies over HTTP instead of
HTTPS.

Signed-off-by: Jonathan Leitschuh <Jonathan.Leitschuh@gmail.com>

solemabrothers committed:
REPORT-861:Make Reports Return an IndicatorResult Datatype (#195)

* https://issues.openmrs.org/browse/REPORT-861:Reports Return a Double Value for Indicator Results

openmrs-bot committed:
[maven-release-plugin] prepare release 1.20.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

dkayiwa committed:
Add support for OpenMRS Platform 2.4 - REPORT-863

mseaton committed:
REPORT-864 - Make EvaluationContext contextValues available to SqlFileDataSetDefinition

Am-Coder committed:
REPORT-866 Properly map the sequences via hibernate (#200)

Am-Coder committed:
REPORT-867 Liquibase error on starting Reporting Module (#201)

mogoodrich committed:
REPORT-865: Support creating Report Configuration via configuration (#199)

mogoodrich committed:
Add dependency on com.fasterxml.jackson.com; switch to version used by OpenMRS Core

dependabot[bot] committed:
Bump jackson-databind from 2.9.8 to 2.9.10.5 (#202)

Bumps [jackson-databind](https://github.com/FasterXML/jackson) from 2.9.8 to 2.9.10.5.
- [Release notes](https://github.com/FasterXML/jackson/releases)
- [Commits](https://github.com/FasterXML/jackson/commits)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mogoodrich committed:
Bump up version of com.fasterxml.jackson.*

mogoodrich committed:
Correct version of com.fasterxml.jackson.*

mogoodrich committed:
add additional valid liquibase checksums

Am-Coder committed:
Report 869  Change tinyint(1) to boolean for boolean columns in Reporting Module (#203)

* REPORT-869 Change tinyint(1) to boolean for boolean columns in Reporting Module

* REPORT-869 Change tinyint(1) to boolean for boolean columns in Reporting Module

dkayiwa committed:
Switching from snapshot to alpha of the platform

openmrs-bot committed:
[maven-release-plugin] prepare release 1.21.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

mseaton committed:
REPORT-872
Cohort Indicator Evaluator does not properly limit aggregation data to numerator cohort

varsha5595 committed:
Adding HTML Encoding for Report Processor Name (#206)

anupamdevulapalli committed:
Added sanitization for EMPT-103 and EMPT-104

isears committed:
Merge pull request #207 from anupamdevulapalli/EMPT-103-and-EMPT-104

 RA-1865: Patch bugs discovered by NCSU team

ibacher committed:
Travis -> GitHub Actions

Feel free to revert, but see https://talk.openmrs.org/t/ci-within-openmrs-bamboo-vs-travis-vs-github-actions-vs-azure-pipelines/31252/6

sherrif10 committed:
REPORT-874:ReportLoaderIntegrationTest Have Failing Tests (#208)

* resolving failing test

* Remove file

* Removed second file

* Removed second file

* https://issues.openmrs.org/browse/REPORT-874

Using File separator method

* Adding spaces btn + signs

dependabot[bot] committed:
Bump jackson-databind from 2.9.10.5 to 2.9.10.7 (#212)

Bumps [jackson-databind](https://github.com/FasterXML/jackson) from 2.9.10.5 to 2.9.10.7.
- [Release notes](https://github.com/FasterXML/jackson/releases)
- [Commits](https://github.com/FasterXML/jackson/commits)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mozzy11 committed:
REPORT-722: Aded a check whether the logic module is installed to handle lack of logic service gracefully. made Reporting module aware of logic modle (#169)

mseaton committed:
REPORT-877: Configurable data set descriptors should support fixed parameter values (#214)

mseaton committed:
REPORT-877: Configurable data set descriptors should support fixed parameter values - attempt to fix unit tests.

openmrs-bot committed:
[maven-release-plugin] prepare release 1.22.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

Parth59 committed:
C:out changes to manageReportDesigns.jsp for xss (#215)

dbaluku committed:
REPORT-878:Enable RenderReport Method to accept multiple values for a parameter passed in a HTTP request (#216)

openmrs-bot committed:
[maven-release-plugin] prepare release 1.23.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

ibacher committed:
Updating repository URLs to HTTPS

For details on why this change is happening see this Talk post: https://talk.openmrs.org/t/maven-3-8-1-and-http-repositories/33364

If this commit causes issues, please feel free to revert

mogoodrich committed:
REPORTING-881: Sql Reports fail if generating user's name has a single quote (#218)

mseaton committed:
REPORT-882 - Support adding Report Processors via config descriptors (#219)

mseaton committed:
Update POM to jfrog urls

mseaton committed:
REPORT-883 - Add new interface and component model for loading data s… (#220)

REPORT-883 - Add new interface and component model for loading data sets from configuration and remove hard-coding for sql-only.

mseaton committed:
Add additional missing method.

mseaton committed:
REPORT-884 - Reporting module should not load reports from config automatically in activator (#221)

mseaton committed:
REPORT-885 - Add ability to refresh a single descriptor

dkayiwa committed:
Adding the 2.4 profile

dkayiwa committed:
Fixing 2.1 profile failing tests

icrc-toliveira committed:
REPORT-880: Added profiles for Openmrs platform 2.3 and 2.4 (#217)

dependabot[bot] committed:
Bump jackson-databind from 2.9.10.7 to 2.9.10.8 (#222)

Bumps [jackson-databind](https://github.com/FasterXML/jackson) from 2.9.10.7 to 2.9.10.8.
- [Release notes](https://github.com/FasterXML/jackson/releases)
- [Commits](https://github.com/FasterXML/jackson/commits)

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson.core:jackson-databind
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

isaaclin-artist committed:
REPORT-492: Multiple stored XSS via Dimension Name and Descriptions (#180)

dkayiwa committed:
REPORT-886 liquibase.util.StringUtils does not exist on platform 2.6

pwargulak committed:
REPORT-870: Increased size of ReportDesignResource.contentType DB column (#227)

- Increased reporting_report_design_resource.content_type to varchar(100),
  so it can fir the XLSX content type value

dkayiwa committed:
REPORT-887 Fix NPE for null DefinitionLibrary.getKeyPrefix()

openmrs-bot committed:
[maven-release-plugin] prepare release 1.24.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

kumuditha-udayanga commented:
Hello @niyolynate, Before working on any changes make sure to take the latest updates from the upstream's master branch to update your remote.
In order to fix this issue, I would suggest :
1. In your local project, create a new branch from the master branch, `git branch feature-{issue number}`.
2. Checkout to the newly created branch `git checkout {branch name}`
3. Reset that branch to upstream's master branch `git reset --hard upstream master`. And pull the latest changes by `git pull upstream master`.
4. Then start working on your changes, or else you can get the changes that you committed in the master branch using `cherry-pick`. (you must copy the checksum of the commit message from the master branch) 
`git cherry-pick {checksum}`
5. If there are any changes before pushing the changes, commit the changes and take a pull from upstream and push the changes to your origin (make sure to push the changes using the new branch name, so it creates a new branch in the origin, and make a pull request using that branch).

alaboso committed:
REPORT-888 - Add MariaDb into changeSet Entries (#232)

Co-authored-by: Amos Laboso <alaboso@uonbi.ac.ke>

alaboso committed:
Bump parent.version to 1.25.0-SNAPSHOT (#233)

dkayiwa committed:
LUI-189 Include suffix in controller urls

icrc-loliveira committed:
REPORT-873: SQL reports to use cursors to iterate through datasets. (#213)

Co-authored-by: Luis Oliveira <luis.oliveira@icrc.org>

anjula-sack committed:
Add the missing api module dependency (#236)

Remove unnecessary dependency

Change the OMRS version for api-1.10

anjula-sack committed:
Update the openmrs version for the api 1.10 (#237)

openmrs-bot committed:
[maven-release-plugin] prepare release 1.25.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

ibacher committed:
Fix POM version

mseaton committed:
REPORT-895 - Remove use of instance variable for connection object (#240)

mogoodrich committed:
REPORT-896: Update Version of Joda Time Library (#243)

dkayiwa committed:
LUI-189 Include suffix in controller urls

mseaton committed:
REPORT-899 ReportLoader should not set parameters to required=false by default (#246)

mseaton committed:
REPORT-900 - HQL Query Builder should support entity names that differ from class names

openmrs-bot committed:
[maven-release-plugin] prepare release 1.26.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

dkayiwa committed:
REPORT-902 Add MariaDb into changeSet Entries

mseaton committed:
REPORT-903 - Enable configuring parameter "widgetConfiguration" from yaml report definitions

dkayiwa committed:
TRUNK-6188: Add whitelisting for components loaded via XStream

dkayiwa committed:
Fix failing build

dkayiwa committed:
TRUNK-6188 fix failing tests

mogoodrich committed:
revert: TRUNK-6188: Add whitelisting for components loaded via Xstream

wikumChamith committed:
TRUNK-6218: Enable XStream whitelisting on OpenMRS Core 2.7.0 and after (#253)

pwargulak committed:
REPORT-906: Add pagination support to ReportRequest read methods (#251)

Also related to O3-1494
---------
Co-authored-by: druchniewicz <druchniewicz@soldevelo.com>

mseaton committed:
TRUNK-6218: Defer setup of xstream security until needed

druchniewicz committed:
REPORT-906: Corrected order of fetching report requests (#254)

openmrs-bot committed:
[maven-release-plugin] prepare release 1.27.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

xprl-gjf committed:
REPORT-909 Fix report renderer paths for editing report designs (#255)

dkayiwa committed:
Update github actions to v4

dkayiwa committed:
Add distribution: 'temurin' to actions/setup-java

dkayiwa committed:
Update version to match with the rest

xprl-gjf committed:
REPORT-908 Unescape XML before deserializing column definition (#259)

mseaton committed:
REPORT-911: Add missing translations

wikumChamith committed:
REPORT-913: Migrate to OpenMRS Platform 2.4.0 (#262)

* Migrate to 2.4.0

* Move api-1.9

* Merge old api modules to the main

* Set the scope of reportingcompatibility to provided

mseaton committed:
Revert "REPORT-913: Migrate to OpenMRS Platform 2.4.0 (#262)"

This reverts commit e4022f6a54ff5f4f2c0ddd0fdaf810a2f0b8d890.

wikumChamith committed:
REPORT-913: Migrate to OpenMRS Platform 2.4.0 (#263)

* REPORT-913: Migrate to OpenMRS Platform 2.4.0 (#262)

* Migrate to 2.4.0

* Move api-1.9

* Merge old api modules to the main

* Set the scope of reportingcompatibility to provided

(cherry picked from commit e4022f6a54ff5f4f2c0ddd0fdaf810a2f0b8d890)

* Remove reporting compatability module

* Remove old datasets

mseaton committed:
REPORT-913: Follow-up fix to exclude previous dependencies

openmrs-bot committed:
[maven-release-plugin] prepare release 1.28.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

rkorytkowski committed:
Constructor must be public for CGLIB proxies, see TRUNK-6382

wikumChamith committed:
REPORT-912: Update to OpenMRS platform 2.7.3 and Support Java 21 (#261)

* Update to Platform 2.7.0 and Support Java 21

* Migrate to org.openmrs.test.jupiter.BaseModuleContextSensitiveTest

* Add the EvaluationProfilerTest24On

Ruhanga committed:
REPORT-912: Added Bamboo specs. (#264)

Ruhanga committed:
REPORT-912: Updated Bamboo spec Development version

Ruhanga committed:
Fixed Bamboo Specs release-scripts repository checkout.

mseaton committed:
REPORT-916 - Support loading custom renderers via config (#265)

mseaton committed:
Upgrade dependencies to non-snapshot for release

mseaton committed:
Update javadoc plugin configuration

openmrs-bot committed:
[maven-release-plugin] prepare release 2.0.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

dkayiwa committed:
Fix com.fasterxml.jackson.databind.ObjectReader LinkageError

openmrs-bot committed:
[maven-release-plugin] prepare release 2.1.0

openmrs-bot committed:
[maven-release-plugin] prepare for next development iteration

wikumChamith committed:
Migrate to Reusable GitHub Action (#268)

ibacher committed:
Separate OWASP dependency check into its own job (#269)

ibacher committed:
Drop obsolete run_owasp_check input and unused secrets

build-backend-module.yml no longer accepts run_owasp_check as an input,
and explicit undeclared secrets fail the workflow. Move the OWASP check
into its own owasp-dependency-check.yml job that only runs on push /
workflow_dispatch, and prune the build job's secrets to just the Maven
credentials it actually consumes.

mseaton committed:
REPORT-359 - Support PDF Report Rendering (#270)
